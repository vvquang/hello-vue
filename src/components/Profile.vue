<template>
  <div class="profile">
    <h1>Profile</h1>
  </div>
</template>

<script>
export default {
  name: "Profile",
  data() {
    return {
      msg: "Profile page"
    };
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h1 {
  font-weight: normal;
}
</style>
