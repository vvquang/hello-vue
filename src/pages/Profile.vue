<template>
  <div class="profile">
    <h1>Profile</h1>
    <div>User Id: {{ $route.params.id }}</div>
    <div>Username: {{ username }}</div>
    <!-- <p>
      <router-link to="/profile/"{{ $route.params.id }}"/edit"
        >Edit profile</router-link
      >
    </p> -->
  </div>
</template>

<script>
export default {
  name: "Profile",
  data: function() {
    return {
      username: "vvquang",
      fullName: "Quang Vu Van"
    };
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h1 {
  font-weight: normal;
}
</style>
