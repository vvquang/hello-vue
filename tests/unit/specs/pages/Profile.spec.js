import { shallowMount } from '@vue/test-utils'
import Profile from '@/pages/Profile'
import wrapperOps from '../../supports/wrapper'

const wrapper = shallowMount(Profile, {
  ...wrapperOps,
  propsData: {
    username: 'user name',
    fullName: 'full name'
  }
})

describe('test Profile.vue', () => {
  it('render app', () => {
    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })
})
