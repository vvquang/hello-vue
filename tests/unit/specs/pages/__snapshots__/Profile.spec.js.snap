// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Profile.vue render app 1`] = `
<div class="profile" username="user name" fullname="full name">
  <h1>Profile</h1>
  <div>User Id: </div>
  <div>Username: vvquang</div>
</div>
`;
